src/BizHawk.Emulation.Cores/Computers/DOS/DOSBox.cs

```diff
@@ -0,0 +1,492 @@
+
+			if (! _cdRomFileToReaderMap.ContainsKey( cdRomName ) ) throw new InvalidOperationException($"Unrecognized CD File with name: {cdRomName}");
+
+            byte[] sectorBuffer = new byte[4096];
+            var cdRomReader = _cdRomFileToReaderMap[cdRomName];
+            switch (sectorSize)
+            {
+				case 2048:
+                    cdRomReader.ReadLBA_2048(lba, sectorBuffer, 0);
+					Marshal.Copy(sectorBuffer, 0, dest, 2048);
+					break;
+                case 2352:
+                    cdRomReader.ReadLBA_2352(lba, sectorBuffer, 0);
+                    Marshal.Copy(sectorBuffer, 0, dest, 2352);
+                    break;
+                case 2448:
+                    cdRomReader.ReadLBA_2448(lba, sectorBuffer, 0);
+                    Marshal.Copy(sectorBuffer, 0, dest, 2448);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unsupported CD sector size: {sectorSize}");
+
+            }
+            DriveLightOn = true;
+        }
+
+        protected override LibWaterboxCore.FrameInfo FrameAdvancePrep(IController controller, bool render, bool rendersound)
+		{
+			DriveLightOn = false;
+			var fi = new LibDOSBox.FrameInfo();
+
+			// Setting joystick inputs
+			fi.joystick1.up      = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Up}") ? 1 : 0;
+			fi.joystick1.down    = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Down}") ? 1 : 0;
+			fi.joystick1.left    = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Left}") ? 1 : 0;
+			fi.joystick1.right   = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Right}") ? 1 : 0;
+			fi.joystick1.button1 = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Button1}") ? 1 : 0;
+			fi.joystick1.button2 = _syncSettings.EnableJoystick1 && controller.IsPressed($"P1 {Inputs.Joystick} {JoystickButtons.Button2}") ? 1 : 0;
+			fi.joystick2.up      = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Up}") ? 1 : 0;
+			fi.joystick2.down    = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Down}") ? 1 : 0;
+			fi.joystick2.left    = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Left}") ? 1 : 0;
+			fi.joystick2.right   = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Right}") ? 1 : 0;
+			fi.joystick2.button1 = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Button1}") ? 1 : 0;
+			fi.joystick2.button2 = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Button2}") ? 1 : 0;

```

Comment:
This might still function if you remove the `_syncSettings.EnableJoystick1` checks, but to keep it fast, wrap the 6 lines in an `if`.